frame/support/procedural/src/pallet/expand/hooks.rs

```diff
@@ -59,6 +58,19 @@ pub fn expand_hooks(def: &mut Def) -> proc_macro2::TokenStream {
 		}
 	};
 
+	let log_try_state = quote::quote! {
+		let pallet_name = <
+			<T as #frame_system::Config>::PalletInfo
+			as
+			#frame_support::traits::PalletInfo
+		>::name::<Self>().unwrap_or("<unknown pallet name>");
```

Comment:
You can use an expect here. We use it everywhere.

```
expect("Every active pallet has a name in the runtime; qed")
```

Reply:
sure.